clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5811,6 +5811,11 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
 
         CmdArgs.push_back("-fsycl-optimize-non-user-code");
       }
+
+      // Disable this option for SYCL by default.
+      // TODO:  This needs to be re-enabled once we have a real fix.
+      CmdArgs.push_back("-fno-offload-use-alloca-addrspace-for-srets");
+
```

Comment:
I think test impact to `clang/test/Driver/sycl-int-footer-old-model.cpp` and `clang/test/Driver/sycl-int-header-footer.cpp` could be avoided by moving this block before line 5684 (before `-fsycl-is-device` is added).

This option is only being added by default for SYCL (and only for device-side compilation). Is it not needed for OpenMP as well? I have vague recollections of OpenMP also being affected, though I could easily be misremembering that.

Reply:
The intent is to add the option for OpenMP in the product compiler not in syclos.

Reply:
@premanandrao, as stated by Tom, adding to above `-fsycl-is-device` will reduce the test updates.  You can also consider moving to before `-fsycl-int-header`.

Reply:
I missed that part in Tom's review.   I moved it to just after `-fsycl-is-device` and that reduced modifying two Driver tests.

Reply:
Another option would be to enable the option for SPIR target. IIRC we need that for OpenMP and OpenCL in downstream. This would help to avoid modifying both. We also need to make sure that native cpu target that is SYCL but doesn't use SPIR is not affected.